plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineStandardWtpProject.java

```diff
@@ -64,43 +55,31 @@ public void execute(IProgressMonitor monitor) throws InvocationTargetException,
         description, "Creating new App Engine Project");
     try {
       operation.execute(progress.newChild(20), uiInfoAdapter);
-      
-      IFacetedProject facetedProject = ProjectFacetsManager.create(
-          newProject, true, progress.newChild(20));
-      
-      installJavaFacet(facetedProject, progress.newChild(10));
-      
       CodeTemplates.materialize(newProject, config, progress.newChild(20));
-      
-      installWebFacet(facetedProject, progress.newChild(10));
-      
-      // must happen after other two facets because the appengine facet requires them
-      FacetInstallDelegate.installAppEngineFacet(facetedProject, progress.newChild(10));
-      FacetInstallDelegate.installAppEngineRuntime(facetedProject, progress.newChild(20));
+
+      Job facetInstallJob = new Job("Install App Engine Facet and runtimes in " + newProject.getName()) {
+
+        @Override
+        protected IStatus run(IProgressMonitor monitor) {
+          try {
+            final IFacetedProject facetedProject = ProjectFacetsManager.create(
+                newProject, true, monitor);
+            AppEngineStandardFacet.installAppEngineFacet(
+                facetedProject, true /* installDependentFacets */, monitor);
+            AppEngineStandardFacet.installAllAppEngineRuntimes(facetedProject, true, monitor);
+          } catch (CoreException e) {
+            return e.getStatus();
+          }
+          return Status.OK_STATUS;
+        }
+  
+      };
+      facetInstallJob.schedule();
```

Comment:
Will the progress monitor behave as expected by the user? The progress for this monitor will be complete on line 81, but the job might still run in the background. Wouldn't this be confusing?